components/script/dom/navigator.rs

```diff
@@ -81,9 +82,55 @@ impl Navigator {
         self.xr.get()
     }
 
-    pub fn gamepads(&self) -> DomRoot<GamepadList> {
-        self.gamepads
-            .or_init(|| GamepadList::new(&self.global(), &[]))
+    pub fn get_gamepad(&self, index: usize) -> Option<DomRoot<Gamepad>> {
+        self.gamepads.borrow().get(index).and_then(|g| g.get())
+    }
+
+    pub fn set_gamepad(&self, index: usize, gamepad: &Gamepad) {
+        if index >= self.gamepads.borrow().len() {
```

Comment:
I think this needs to be moved to inside `select_gamepad_index`(in the spec that is also where you see "Append null to ...").